egs/wsj/s5/steps/nnet3/components.py

```diff
@@ -344,7 +344,261 @@ def AddClstmLayer(config_lines,
     return {
             'descriptor': output_descriptor,
             'dimension':output_dim
-            }
+           }
```

Comment:
Looks good.. Only suggestion would be to use small letter for diagonal components

i.e., instead of W_ic use w_ic..

Also according to Dan's suggestion you can eliminate the BLSTM component and instead just use 2LSTM components one with forward and one with backward connections. And combine their output descriptors using "Append".

Sorry for the rework..